examples/src/bin/stopwatch.rs

```diff
@@ -0,0 +1,115 @@
+use cursive::traits::*;
+
+fn main() {
+    let mut siv = cursive::default();
+    let timer = Timer::TimerView::new();
+    siv.add_layer(timer.fixed_width(8).with_name("timer"));
+    siv.add_layer(cursive::views::Dialog::info(
+        "Press 'Space' to start/pause/resume the timer!",
+    ));
+    siv.set_fps(5);
+    siv.run();
+}
+
+mod Timer {
+    use chrono::{DateTime, Duration, Local};
+    use cursive::{
+        event::{Callback, Event, EventResult, Key},
+        view::View,
+        Cursive, Printer,
+    };
+    use std::rc::Rc;
+
+    pub struct TimerView {
+        elapsed: Duration,
+        last_update: DateTime<Local>,
+        paused: bool,
+        on_stop: Option<Rc<dyn Fn(&mut Cursive, Duration)>>,
+    }
+
+    impl TimerView {
+        pub fn new() -> Self {
+            Self {
+                elapsed: Duration::zero(),
+                last_update: Local::now(),
+                on_stop: None,
+                paused: true,
+            }
+        }
+        pub fn pause(&mut self) {
+            assert!(self.paused == false, "Already paused!");
+            self.elapsed = self.read();
+            self.paused = true;
+        }
+        pub fn resume(&mut self) {
+            assert!(self.paused == true, "Already running!");
+            self.last_update = Local::now();
+            self.paused = false;
+        }
+        pub fn pause_or_resume(&mut self) {
+            if self.paused {
+                self.resume();
+            } else {
+                self.pause();
+            }
+        }
+        pub fn read(&self) -> Duration {
+            if self.paused {
+                self.elapsed
+            } else {
+                self.elapsed + (Local::now() - self.last_update)
+            }
+        }
+
+        /// Sets a callback to be used when `<Enter>` is pressed.
+        ///
+        /// Also happens if the user clicks an item.
+        ///
+        /// The elapsed time will be given to the callback.
+        ///
+        /// See also cursive::views::select_view::SelectView::on_submit
+        pub fn on_stop<F, R>(&mut self, cb: F)
```

Comment:
The usual convention for views would be to have `fn set_on_stop(&mut self, ...)`, and a chainable `fn on_stop(self, ...) -> Self`.

Reply:
Thanks! I've made both `set_on_stop` and `on_stop`.